docker-compose-full-services.yml

```diff
@@ -0,0 +1,130 @@
+  # Mix.Task.run_task/3\n    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2\n    (elixir) lib/code.ex:767:
+  # Code.require_file/2\n'
+  #
+  # watcher_1            | 2019-06-19 10:21:05,641 CRITICAL:Could not update the Postgres database Exiting.
+  #
+  postgres:
+    image: postgres
+    restart: always
+    ports:
+      - "5433:5432"
+    environment:
+      POSTGRES_USER: omisego_dev
+      POSTGRES_PASSWORD: omisego_dev
+      POSTGRES_DB: omisego_dev
+    healthcheck:
+      test: pg_isready -U omisego_dev
+      interval: 5s
+      timeout: 3s
+      retries: 5
+
+  geth:
+    image: ethereum/client-go:v1.8.21
+    entrypoint: /bin/sh -c "apk add curl && geth --targetgaslimit "6200000" --dev --dev.period 1 --rpc --rpcapi personal,web3,eth,net --rpcaddr 0.0.0.0 --rpcvhosts=* --rpcport=8545"
+    ports:
+     - "8545:8545"
+    healthcheck:
+      test: curl geth:8545
+      interval: 5s
+      timeout: 3s
+      retries: 5
+
+  # TODO(jacob): this exists in the mac version of the file, but not the non-mac one. understand why and
+  # check whether this should be in this file or not.
+  elixir-omg:
+    build: .
+    image: elixir-omg:dockercompose
+
+  contractexchanger:
+    image: jakebunce/contractexchanger:v0.0.2
+    ports:
+      - "5000:5000"
+    healthcheck:
+      test: curl contractexchanger:5000
+      interval: 5s
+      timeout: 3s
+      retries: 5
+    depends_on:
+      geth:
+        condition: service_healthy
+
+  childchain:
+    image: elixir-omg:dockercompose
+    entrypoint: /bin/bash -c "./launcher.py && elixir --erl '-sname childchain' -S mix xomg.child_chain.start --config ~/config.exs"
+    environment:
+      - MIX_ENV=dev
+      - EXIT_PERIOD_SECONDS=86400
+      - ELIXIR_SERVICE=CHILDCHAIN
+      - CONTRACT_EXCHANGER_URL=http://contractexchanger:5000
+      - ETHEREUM_RPC_URL=http://geth:8545
+      - CHILD_CHAIN_URL=http://childchain:9656
+      - ETHEREUM_NETWORK=LOCALCHAIN
+    restart: always
+    ports:
+      - "9656:9656"
+    healthcheck:
+      test: curl childchain:9656
+      interval: 10s
+      timeout: 3s
+      retries: 7
+    depends_on:
+      geth:
+        condition: service_healthy
+      contractexchanger:
+        condition: service_healthy
+
+  watcher:
+    image: elixir-omg:dockercompose
+    entrypoint: /bin/bash -c "./launcher.py && elixir --erl '-sname watcher' -S mix xomg.watcher.start --convenience --config ~/config_watcher.exs "
+    environment:
+      - MIX_ENV=dev
+      - ELIXIR_SERVICE=WATCHER
+      - CONTRACT_EXCHANGER_URL=http://contractexchanger:5000
+      - ETHEREUM_RPC_URL=http://geth:8545
+      - CHILD_CHAIN_URL=http://childchain:9656
+      - ETHEREUM_NETWORK=LOCALCHAIN
+      - DATABASE_URL=postgres://omisego_dev:omisego_dev@postgres:5433/omisego_dev
+    restart: always
+    ports:
+      - "7434:7434"
+    healthcheck:
+      test: curl watcher:7434
+      interval: 5s
+      timeout: 3s
+      retries: 5
+    depends_on:
```

Comment:
```  
depends_on:
      postgres:
        condition: service_healthy
      childchain:
        condition: service_healthy
```
even if you add this I still get
```
watcher_1            | 2019-06-19 11:50:21,768 CRITICAL:Could not migrate the database. Error: b'\n11:50:18.820 [error] Postgrex.Protocol (#PID<0.372.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (postgres:5433): connection refused - :econnrefused\n\n11:50:18.820 [error] Postgrex.Protocol (#PID<0.371.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (postgres:5433): connection refused - :econnrefused\n\n11:50:19.977 [error] Postgrex.Protocol (#PID<0.371.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (postgres:5433): connection refused - :econnrefused\n\n11:50:20.447 [error] Postgrex.Protocol (#PID<0.372.0>) failed to connect: ** (DBConnection.ConnectionError) tcp connect (postgres:5433): connection refused - :econnrefused\n\n11:50:21.753 [error] Could not create schema migrations table. This error usually happens due to the following:\n\n  * The database does not exist\n  * The "schema_migrations" table, which Ecto uses for managing\n    migrations, was defined by another library\n  * There is a deadlock while migrating (such as using concurrent\n    indexes with a migration_lock)\n\nTo fix the first issue, run "mix ecto.create".\n\nTo address the second, you can run "mix ecto.drop" followed by\n"mix ecto.create". Alternatively you may configure Ecto to use\nanother table for managing migrations:\n\n    config :omg_watcher, OMG.Watcher.DB.Repo,\n      migration_source: "some_other_table_for_schema_migrations"\n\nThe full error report is shown below.\n\n** (DBConnection.ConnectionError) connection not available and request was dropped from queue after 2968ms. You can configure how long requests wait in the queue using :queue_target and :queue_interval. See DBConnection.start_link/2 for more information\n    (ecto_sql) lib/ecto/adapters/sql.ex:618: Ecto.Adapters.SQL.raise_sql_call_error/1\n    (elixir) lib/enum.ex:1327: Enum."-map/2-lists^map/1-0-"/2\n    (ecto_sql) lib/ecto/adapters/sql.ex:705: Ecto.Adapters.SQL.execute_ddl/4\n    (ecto_sql) lib/ecto/migrator.ex:558: Ecto.Migrator.verbose_schema_migration/3\n    (ecto_sql) lib/ecto/migrator.ex:411: Ecto.Migrator.lock_for_migrations/3\n    (ecto_sql) lib/ecto/migrator.ex:335: Ecto.Migrator.run/4\n    (ecto_sql) lib/ecto/migrator.ex:83: Ecto.Migrator.with_repo/3\n    (ecto_sql) lib/mix/tasks/ecto.migrate.ex:118: anonymous fn/5 in Mix.Tasks.Ecto.Migrate.run/2\n    (elixir) lib/enum.ex:1940: Enum."-reduce/3-lists^foldl/2-0-"/3\n    (ecto_sql) lib/mix/tasks/ecto.migrate.ex:106: Mix.Tasks.Ecto.Migrate.run/2\n    (mix) lib/mix/task.ex:331: Mix.Task.run_task/3\n    (mix) lib/mix/cli.ex:79: Mix.CLI.run_task/2\n    (elixir) lib/code.ex:767: Code.require_file/2\n'
```
probably means watcher can't reach the service... (this is on linux though)